lib/Doctrine/Migrations/Tools/Console/ConsoleInputMigratorConfigurationFactory.php

```diff
@@ -21,12 +22,26 @@ public function getMigratorConfiguration(InputInterface $input): MigratorConfigu
     {
         $timeAllQueries = $input->hasOption('query-time') ? (bool) $input->getOption('query-time') : false;
         $dryRun         = $input->hasOption('dry-run') ? (bool) $input->getOption('dry-run') : false;
-        $allOrNothing   = $input->hasOption('all-or-nothing') ? $input->getOption('all-or-nothing') : null;
-        $allOrNothing   = (bool) ($allOrNothing ?? $this->configuration->isAllOrNothing());
+        $allOrNothing   = $this->determineAllOrNothingValueFrom($input) ?? $this->configuration->isAllOrNothing();
 
         return (new MigratorConfiguration())
             ->setDryRun($dryRun)
             ->setTimeAllQueries($timeAllQueries)
             ->setAllOrNothing($allOrNothing);
     }
+
+    private function determineAllOrNothingValueFrom(InputInterface $input): ?bool
+    {
+        try {
+            $allOrNothingOption = $input->getOption('all-or-nothing');
+        } catch (InvalidArgumentException $exception) {
```

Comment:
For which cases is this necessary? I have an uneasy feeling about this… if it's not strictly related to the issue at hand, I'd remove it.

Reply:
There are some tests that fail if we don't catch it... I found it weird as well, but I would not know how to proceed further otherwise.

The tests that failed when not catching the exception:

- [testWriteSql](https://github.com/doctrine/migrations/blob/3.5.x/tests/Doctrine/Migrations/Tests/Tools/Console/Command/ExecuteCommandTest.php#L50)
- [testExecute](https://github.com/doctrine/migrations/blob/3.5.x/tests/Doctrine/Migrations/Tests/Tools/Console/Command/ExecuteCommandTest.php#L101)
- [testExecuteMultiple](https://github.com/doctrine/migrations/blob/3.5.x/tests/Doctrine/Migrations/Tests/Tools/Console/Command/ExecuteCommandTest.php#L123)
- [testExecuteCancel](https://github.com/doctrine/migrations/blob/3.5.x/tests/Doctrine/Migrations/Tests/Tools/Console/Command/ExecuteCommandTest.php#L155)

Reply:
I think this class your changing is called for several commands. IMO you should remove this `try/catch`, and resort to `hasOption` here, to determine if the command that is called has that option. If it doesn't, then I guess we can omit calling `setAllOrNothing`, or call it with `false`, whatever is the most simple for you.

Reply:
The last push I did seems to be passing all tests. Using the `hasOption` did help.